python/cudf/cudf/core/reshape.py

```diff
@@ -35,16 +35,20 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     if not objs:
         raise ValueError("Need at least one object to concatenate")
 
-    # no-op for single object
-    if len(objs) == 1:
-        return objs[0]
-
     # convert any RangeIndex
     objs = [
         as_index(obj._values) if isinstance(obj, RangeIndex) else obj
```

Comment:
If there is only one `RangeIndex` input we'll be materializing here, as opposed to returning same RangeIndex previously.
If there is are multiple `RangeIndex` inputs we'll be materializing here, same as previous behavior.

@shwina Are we okay with such behavior?

Reply:
Why do we even need to convert RangeIndexes into non-RangeIndexes here? It will just eventually hit `RangeIndex._concat` which should just hit the generic `Index._concat` no?

Reply:
I'm not sure I understand. Is the following going to fail?

```
assert isinstance(
    cudf.concat([cudf.DataFrame({'a': [1, 2, 3]})]).index,
    cudf.core.index.RangeIndex,
)
```

Reply:
@shwina That is not going to fail.
```python
assert isinstance(cudf.concat([cudf.DataFrame({'a': [1, 2, 3]}).index]), cudf.core.index.RangeIndex)
```
This was the case, which will also pass now.
@kkraus14 Yes we are actually materializing in `Index._concat` and there is no need to specially handle this ahead inflow, hence removed the check.